### ISSUE TITLE
[Bug]: MongoDB::Atlas::DatabaseUser - drift detection not working - read handler does not return "AWSIAMType" and "Scopes" properties

### ISSUE THREAD

**Issue by StefanWachter1507** [mongodb/mongodbatlas-cloudformation-resources#1233]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### CFN Resource version

v2.1.0

### CFN Resource Region

eu-central-1

### Current Behavior

CFN drift detection detects a drift:

![Screenshot from 2024-11-27 13-20-55](https://github.com/user-attachments/assets/f309ab7e-82a9-4c2f-baf2-ce983115cc3c)

The "AWSIAMType" and "Scopes" properties are missing.

The Read handler of the [MongoDB::Atlas::DatabaseUser](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/database-user/cmd/resource/resource.go#L86) resource does not process the "Scopes" property. In addition, the condition for copying the "AWSIAMType" property seems to be erronous. Why should that property only be copied, if some default value has been set???

```
	if currentModel.AWSIAMType != nil {
		currentModel.AWSIAMType = databaseUser.AwsIAMType
	}
```

### CFN template to reproduce the issue

```yaml
{
      "Type": "MongoDB::Atlas::DatabaseUser",
      "Properties": {
        "AWSIAMType": "ROLE",
        "DatabaseName": "$external",
        "ProjectId": "...",
        "Roles": [
          {
            "DatabaseName": "...",
            "RoleName": "read"
          }
        ],
        "Scopes": [
          {
            "Name": "...",
            "Type": "CLUSTER"
          }
        ],
        "Username": "..."
      }
```

### Steps To Reproduce

Deploy DatabaseUser resource in a stack and execute drift detection on that stack.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by EspenAlbert** [comment]
Hi @StefanWachter1507 .
Thank you for reporting the issue.
I was able to reproduce and create a fix in #1234 
Hopefully, we make a release by the end of the week.

**Comment by Zuhairahmed** [comment]
hi @StefanWachter1507 just quick update we are still working on this issue. @EspenAlbert can you share a revised ETA? thanks!

**Comment by EspenAlbert** [comment]
Hi. Sorry for the delay.
I just released this in `MongoDB-Atlas-DatabaseUser/2.2.0`
Please try it out and re-open if it doesn't work 👍
